Automattic/jetpack
# Social: Add connection management tests

Fixes https://github.com/Automattic/jetpack-reach/issues/360

## Proposed changes:

* Adds unit tests for components
* Adds some general integration test for the connection list

### Other information:


## Jetpack product discussion


https://github.com/Automattic/jetpack-reach/issues/360
## Does this pull request change what data or activity we track or use?


No
## Testing instructions:






* Go to the publicize components folder `projects/js-packages/publicize-components`
* Run tests with `pnpm run test`

  ConnectionStatus
    ✓ renders nothing when connection status is "ok" (19 ms)
    ✓ renders nothing when connection status is undefined (2 ms)
    ✓ renders the appropriate notice and Reconnect component when connection status is not "ok" or undefined (4 ms)
    ✓ renders the correct notice when connection status is "refresh-failed" (2 ms)

  Reconnect
    ✓ renders the Reconnect button with correct label (35 ms)
    ✓ disables the button when isDisconnecting is true (9 ms)
    ✓ calls deleteConnectionById and requestAccess on button click (23 ms)
    ✓ does not render the button if connection cannot be disconnected (2 ms)

  Marking a connection as shared
    ✓ toggling the checkbox should mark a connection as shared (135 ms)
    ✓ toggle is disabled while updating (8 ms)

  Disconnecting a connection
    ✓ disconnecting a connection should call the disconnect method (200 ms)
    ✓ panel is disabled while updating (11 ms)
    ✓ button changes name and is disabled while updating (8 ms)

  ConnectionManagement
    No connections
      ✓ renders the component with no connections (43 ms)
    With connections
      ✓ renders the spinner without connection name (36 ms)
      ✓ renders the component with proper connections (13 ms)
      Connection panel
        ✓ connection panel can be opened and closed (111 ms)
        ✓ clicking disconnect should bring up confirmation which can be cancelled (96 ms)
        ✓ mark as shared component is visible if user has permission (43 ms)
